api/instance/v1/server_utils.go

```diff
@@ -127,3 +132,109 @@ func (s *API) GetServerType(req *GetServerTypeRequest) (*ServerType, error) {
 
 	return nil, errors.New("could not find server type %q", req.Name)
 }
+
+// GetServerUserDataRequest is used by GetServerType method.
+type GetServerUserDataRequest struct {
+	Zone     utils.Zone `json:"-"`
+	ServerID string     `json:"-"`
+
+	// Key define the user data key to get
+	Key string `json:"-"`
+}
+
+// GetServerUserData get user data
+//
+// Get the content of a user data with the given key on a server
+func (s *API) GetServerUserData(req *GetServerUserDataRequest, opts ...scw.RequestOption) (io.ReadCloser, error) {
+	var err error
+
+	if req.Zone == "" {
+		defaultZone, _ := s.client.GetDefaultZone()
+		req.Zone = defaultZone
+	}
+
+	if fmt.Sprint(req.Zone) == "" {
+		return nil, errors.New("field Zone cannot be empty in request")
+	}
+
+	if fmt.Sprint(req.ServerID) == "" {
+		return nil, errors.New("field ServerID cannot be empty in request")
+	}
+
+	if fmt.Sprint(req.Key) == "" {
+		return nil, errors.New("field Key cannot be empty in request")
+	}
+
+	scwReq := &scw.ScalewayRequest{
+		Method:  "GET",
+		Path:    "/instance/v1/zones/" + fmt.Sprint(req.Zone) + "/servers/" + fmt.Sprint(req.ServerID) + "/user_data/" + fmt.Sprint(req.Key),
+		Headers: http.Header{},
+	}
+
+	buffer := &bytes.Buffer{}
+
+	err = s.client.Do(scwReq, buffer, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	res := ioutil.NopCloser(buffer)
+	return res, nil
+}
+
+type SetServerUserDataRequest struct {
+	Zone     utils.Zone `json:"-"`
+	ServerID string     `json:"-"`
+
+	// Key define the user data key to set
+	Key string `json:"-"`
+
+	// Content define the data to set
+	Content io.Reader
+}
+
+// SetServerUserData add/Set user data
```

Comment:
let's kill the ambiguity between `add/Set` ==> `SetServerUserData overwrites the user data associated to a server`